### ISSUE TITLE
`amplify pull` replaces hooks directory with contents from deployment bucket

### ISSUE THREAD

**Issue by josefaidt** [aws-amplify/amplify-cli#10253]
### Before opening, please confirm:

- [X] I have installed the latest version of the Amplify CLI (see above), and confirmed that the issue still persists.
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-cli/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.
- [X] I have removed any sensitive information from my code snippets and submission.

### How did you install the Amplify CLI?

pnpm

### If applicable, what version of Node.js are you using?

16.14.2

### Amplify CLI Version

8.0.2

### What operating system are you using?

mac

### Did you make any manual changes to the cloud resources managed by Amplify? Please describe the changes made.

no

### Amplify Categories

Not applicable

### Amplify Commands

pull

### Describe the bug

When making changes to command hooks locally and running `amplify pull`, the CLI removes/reverts changes to the hooks. In a CI/CD setting, running a headless `amplify pull` command will remove the hooks directory entirely and replace with the contents from the deployment bucket, which removes installed dependencies (i.e. `node_modules` directory).

#### Related issues:

- https://github.com/aws-amplify/amplify-hosting/issues/2492
- https://github.com/aws-amplify/amplify-hosting/issues/2624

### Expected behavior

- command hooks are not uploaded to the deployment bucket OR treat as a backend resource where these files are not overwritten on pull
- headless `amplify pull` should not replace the hooks directory in an _uninitialized_ project
- `amplify pull` should not revert changes/additions in an _initialized_ project

### `amplify pull`

1. create a project with `amplify init -y`
2. add a `pre-push.js` command hook
	```js
    console.log('hello')
    setTimeout(() => {
      console.log('pre-push')
    }, 2000)
	```
3. push with `amplify push -y`
4. add another hook `post-push.js`
	```js
    console.log('hello')
    setTimeout(() => {
      console.log('post-push')
    }, 2000)
	```
5. run `amplify pull`
6. observe CLI removes newly added `post-push.js` hook

### headless `amplify pull` (CI/CD)

Using the example above, add the following files:

- `amplify/hooks/package.json`
	```json
    {
      "type": "module",
      "dependencies": {
        "zx": "^6.1.0"
      }
    }
	```
- `amplify/hooks/hooks-config.json`
	```json
    {
      "ignore": ["node_modules"]
    }
	```
- `pull.sh`
	```bash
    #!/bin/bash
    set -e
    IFS='|'

    AMPLIFY_APP_ID='d1lob39t6eexky' # replace with Amplify App ID
    AMPLIFY_ENVIRONMENT='dev'
    AWS_REGION='us-east-1'

    AWSCLOUDFORMATIONCONFIG="{\
    \"configLevel\":\"project\",\
    \"useProfile\":true,\
    \"profileName\":\"default\",\
    \"region\":\"$AWS_REGION\"\
    }"
    AMPLIFY="{\
    \"appId\":\"$AMPLIFY_APP_ID\",\
    \"envName\":\"$AMPLIFY_ENVIRONMENT\",\
    }"

    PROVIDERS="{\
    \"awscloudformation\":$AWSCLOUDFORMATIONCONFIG}"

    amplify pull \
      --amplify $AMPLIFY \
      --providers $PROVIDERS \
      --yes || true
	```

1. push the project to a git repository (NOTE: you will not experience the following behavior if the project is already initialized)
2. download the git repository's contents, unzip
3. install dependencies in hooks folder: `cd amplify/hooks; npm i; cd -`
4. run the pull script with `bash pull.sh`
5. observe hooks directory is removed during the pull operation and is ultimately replaced with contents from the deployment bucket

### GraphQL schema(s)

<details>

```graphql
# Put schemas below this line


```

</details>

### Log output

<details>

```
# Put your logs below this line


```

</details>

### Additional information

To workaround this issue we can opt-out of using Amplify Hosting's helper script in favor of using our own [headless init](https://docs.amplify.aws/cli/usage/headless/#sample-script) script. In short, we can use the headless pull script in the CI/CD scenario above as an init script (note: for simplicity of the example we're hardcoding environment name and region, however these can be passed in as environment variables or retrieved from the `amplify/team-provider-info.json` file):

```bash
#!/bin/bash
set -e
IFS='|'

AMPLIFY_ENVIRONMENT='dev'
AWS_REGION='us-east-1'

AWSCLOUDFORMATIONCONFIG="{\
\"configLevel\":\"project\",\
\"useProfile\":true,\
\"profileName\":\"default\",\
\"region\":\"$AWS_REGION\"\
}"
AMPLIFY="{\
\"envName\":\"$AMPLIFY_ENVIRONMENT\",\
}"

PROVIDERS="{\
\"awscloudformation\":$AWSCLOUDFORMATIONCONFIG}"

amplify init \
  --amplify $AMPLIFY \
  --providers $PROVIDERS \
  --yes || true
```

And modify our buildspec (`amplify.yml`) to use the headless init script:

```yaml
version: 1
backend:
  phases:
    build:
      commands:
        - cd amplify/hooks; npm i; cd -
        - sh init.sh
        - amplify push -y
frontend:
  phases:
    # IMPORTANT - Please verify your build commands
    build:
      commands: []
  artifacts:
    # IMPORTANT - Please verify your build output directory
    baseDirectory: /
    files:
      - '**/*'
  cache:
    paths: []
```

**Comment by lazpavel** [comment]
docs changes PR related to this work can be found here: https://github.com/aws-amplify/docs/pull/4458

**Comment by lazpavel** [comment]
Issue fixed in v10.5.0
